src/bin/taker.rs

```diff
@@ -1,3 +1,183 @@
+use std::path::PathBuf;
+
+use bitcoind::bitcoincore_rpc::{json::ListUnspentResultEntry, Auth};
+use clap::Parser;
+use coinswap::{
+    taker::{SwapParams, Taker, TakerBehavior},
+    utill::{
+        parse_proxy_auth, read_bitcoin_network_string, read_connection_network_string, setup_logger,
+    },
+    wallet::RPCConfig,
+};
+
+#[derive(Parser, Debug)]
+#[clap(version = option_env ! ("CARGO_PKG_VERSION").unwrap_or("unknown"),
+author = option_env ! ("CARGO_PKG_AUTHORS").unwrap_or(""))]
+struct Cli {
+    #[clap(long, default_value = "clearnet",possible_values = &["tor","clearnet"])]
+    network: String,
+    #[clap(long, short = 'd')]
+    data_directory: Option<PathBuf>,
+    #[clap(
+        name = "ADDRESS:PORT",
+        long,
+        short = 'r',
+        default_value = "127.0.0.1:18444"
+    )]
+    pub rpc: String,
+    #[clap(name="USER:PASSWORD",short='a',long, value_parser = parse_proxy_auth, default_value = "user:password")]
+    pub auth: (String, String),
+    #[clap(
+        name = "NETWORK",
+        long,
+        short = 'n',
+        default_value = "regtest", possible_values = &["regtest", "signet", "mainnet"]
+    )]
+    pub rpc_network: String,
+    #[clap(name = "WALLET", long, short = 'w', default_value = "taker")]
+    pub wallet_name: String,
+    #[clap(name = "maker_count", default_value = "2")]
+    pub maker_count: u16,
+    #[clap(name = "send_amount", default_value = "500000")]
+    pub send_amount: u64,
+    #[clap(name = "tx_count", default_value = "3")]
+    pub tx_count: u32,
+    #[clap(name = "fee_rate", default_value = "1000")]
+    pub fee_rate: u64,
+    #[clap(name = "required_confirms", default_value = "1000")]
+    pub required_confirms: u64,
+    #[clap(subcommand)]
+    command: Commands,
```

Comment:
How about instead of these params in the args, we take in a config file_path. And there the user can specify all the values for swap params and many more. I feel that could be more intuitive than a long list of args in cli-app. What say?
 
 Can be a followup PR. It doesn't have to be in this.

Reply:
Potential New Issue